cumulus/parachain-template/runtime/src/impl.rs

```diff
@@ -434,6 +437,17 @@ impl_runtime_apis! {
 			let _ = Executive::apply_extrinsic(extrinsic);
 			Executive::storage_root()
 		}
+
+		fn construct_set_code_extrinsic(code: Vec<u8>) -> Vec<u8> {
```

Comment:
Can't we return unchecked extrinsic itself? It will be encoded when crossing runtime boundary anyway.

Reply:
We can't as the runtime is unaware of any concrete types.